azurecosmos/src/main/java/com/yahoo/ycsb/db/AzureCosmosClient.java

```diff
@@ -0,0 +1,454 @@
+
+    this.databaseName = this.getStringProperty(
+            "azurecosmos.databaseName",
+            DEFAULT_DATABASE_NAME);
+
+    this.maxDegreeOfParallelismForQuery = this.getIntProperty(
+            "azurecosmos.maxDegreeOfParallelismForQuery",
+            DEFAULT_MAX_DEGREE_OF_PARALLELISM_FOR_QUERY);
+
+    this.includeExceptionStackInLog = this.getBooleanProperty(
+            "azurecosmos.includeExceptionStackInLog",
+            DEFAULT_INCLUDE_EXCEPTION_STACK_IN_LOG);
+
+    ConsistencyLevel consistencyLevel = ConsistencyLevel.valueOf(this.getStringProperty(
+            "azurecosmos.consistencyLevel",
+            DEFAULT_CONSISTENCY_LEVEL));
+    String connectionModeString = this.getStringProperty(
+            "azurecosmos.connectionMode",
+            DEFAULT_CONNECTION_MODE);
+
+    ConnectionPolicy connectionPolicy = new ConnectionPolicy();
+    connectionPolicy.setEnableEndpointDiscovery(false);
+    connectionPolicy.setConnectionMode(ConnectionMode.valueOf(connectionModeString));
+    connectionPolicy.setMaxPoolSize(this.getIntProperty("azurecosmos.maxConnectionPoolSize",
+                                                        connectionPolicy.getMaxPoolSize()));
+    connectionPolicy.setIdleConnectionTimeout(this.getIntProperty("azurecosmos.idleConnectionTimeout",
+                                                                  connectionPolicy.getIdleConnectionTimeout()));
+
+    RetryOptions retryOptions = new RetryOptions();
+    retryOptions.setMaxRetryAttemptsOnThrottledRequests(this.getIntProperty(
+            "azurecosmos.maxRetryAttemptsOnThrottledRequests",
+            retryOptions.getMaxRetryAttemptsOnThrottledRequests()));
+    retryOptions.setMaxRetryWaitTimeInSeconds(this.getIntProperty(
+              "azurecosmos.maxRetryWaitTimeInSeconds",
+              retryOptions.getMaxRetryWaitTimeInSeconds()));
+    connectionPolicy.setRetryOptions(retryOptions);
+
+    // Query text
+    this.queryText = this.getQueryText();
+
+    try {
+      LOGGER.info("Creating azurecosmos client {}.. connectivityMode={}, consistencyLevel={},"
+                      + " maxRetryAttemptsOnThrottledRequests={}, maxRetryWaitTimeInSeconds={}"
+                      + " useSinglePartitionCollection={}, useUpsert={}, useHashQueryForScan={}, "
+                      + "queryText={}",
+              uri,
+              connectionPolicy.getConnectionMode(),
+              consistencyLevel.toString(),
+              connectionPolicy.getRetryOptions().getMaxRetryAttemptsOnThrottledRequests(),
+              connectionPolicy.getRetryOptions().getMaxRetryWaitTimeInSeconds(),
+              this.useSinglePartitionCollection,
+              this.useUpsert,
+              this.useHashQueryForScan,
+              this.queryText);
+      AzureCosmosClient.client = new DocumentClient(uri, primaryKey, connectionPolicy, consistencyLevel);
+      LOGGER.info("Azure Cosmos connection created: {}", uri);
+    } catch (Exception e) {
```

Comment:
Here and below, is there a more specific Exception (as supported by the Azure Cosmos Java Client) that  we can catch for the specific error condition we are expecting here?

Reply:
exceptions are not documented.  Feedback set to the team
https://docs.microsoft.com/en-us/java/api/com.microsoft.azure.documentdb._document_client?view=azure-java-stable

Even so I looked at the SDK code and updated.